### PR TITLE
[Workflow] Added parameter typhint where possible

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | mas
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | ye
| Fixed tickets | partially #32179
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php [hunk 1]
```diff
@@ -236,15 +236,15 @@ protected function endDot()
...
         return hash('sha1', $id);
     }
 
     /**
      * @internal
      */
-    protected function escape($value): string
+    protected function escape(string $value): string
     {
         return \is_bool($value) ? ($value ? '1' : '0') : addslashes($value);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php]: not sure about the type hint, give the code

Comment (lyrixx) [reply src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php]: Good catch! Fixed

File: src/Symfony/Component/Workflow/DefinitionBuilder.php [hunk 2]
```diff
@@ -61,8 +61,6 @@ public function clear()
     }
 
     /**
-     * @param string $place
-     *
      * @return $this
      */
     public function setInitialPlace($place)
```
Comment (derrabus) [inline src/Symfony/Component/Workflow/DefinitionBuilder.php]: Did you forget to add the type-hint here?

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Workflow/DefinitionBuilder.php (hunk 2 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: :thinking: actually, this is more complicated. This param was wrong. We forgot to update it / deprecate this method / add a new one. Now we accept an array of initial places.

I forgot that in #30468 

I will open a new PR to fix this. This change is valid though

Comment (derrabus) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: In that case we shouldn‘t remove the `@param` annotation but replace it with a more appropriate one.

Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: Thanks again, here we go https://github.com/symfony/symfony/pull/32258

Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: Let's block this PR until the other is merged. Then I will be able to drop this method in master :)

Comment (fabpot) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: #32258 has been merged now

Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: Rebased, :+1: Thanks

**Comment by lyrixx** [comment]
Are you OK with this one?  cc @derrabus

File: src/Symfony/Component/Workflow/Dumper/DumperInterface.php [hunk 3]
```diff
@@ -25,10 +25,6 @@ interface DumperInterface
     /**
      * Dumps a workflow definition.
      *
-     * @param Definition   $definition A Definition instance
-     * @param Marking|null $marking    A Marking instance
-     * @param array        $options    An array of options
-     *
```
Comment (derrabus) [inline src/Symfony/Component/Workflow/Dumper/DumperInterface.php]: This should be done on a lower branch.

File: src/Symfony/Component/Workflow/Event/GuardEvent.php [hunk 4]
```diff
@@ -27,9 +28,9 @@ class GuardEvent extends Event
     /**
      * {@inheritdoc}
      */
-    public function __construct($subject, Marking $marking, Transition $transition, $workflowName = null)
```
Comment (derrabus) [inline src/Symfony/Component/Workflow/Event/GuardEvent.php]: The `$workflowName` => `WorkflowInterface $workflow = null` change should be done on a lower branch.

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Workflow/Dumper/DumperInterface.php (hunk 3 above)
Comment (fabpot) [reply src/Symfony/Component/Workflow/Dumper/DumperInterface.php]: Let's minimize the changes that don't add value to the code. Keeping that in lower branches is fine by me.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @lyrixx.

**Pull request merged**